### ISSUE TITLE
[Package Request] - geos

### ISSUE THREAD

**Issue by stangelandcl** [amazonlinux/amazon-linux-2023#372]
**What package is missing from Amazon Linux 2023? Please describe and include package name.**
geos and geos-devel. C/C++ port of JTS for manipulating geometric shapes particularly for GIS. https://libgeos.org

**Is this an update to existing package or new package request?**
new package request

**Is this package available in Amazon Linux 2? If it is available via external sources such as EPEL, please specify.**
It is available in Fedora. 
https://packages.fedoraproject.org/pkgs/geos/geos-devel/
https://packages.fedoraproject.org/pkgs/geos/geos/

**Any additional information you'd like to include. (use-cases, etc)**
It is a commonly used library for manipulating geographic shapes. It is a dependency of Python's shapely, R's sf libraries, and Postgres' PostGIS.

**Comment by OpenSourceable** [comment]
+1

**Comment by Raju-perumal** [comment]
+1

I am also looking for the postgis packages.

-- postgis33_15-3.3.3*
-- postgis33_15-utils-3.3.3*

Please keep us updated with status.

Thanks,

**Comment by rooby** [comment]
You might be best to create a separate issue about that @Raju-perumal

**Comment by Raju-perumal** [comment]
Yes Created one : https://github.com/amazonlinux/amazon-linux-2023/issues/730

**Comment by januszm** [comment]
Is there any progress on this? GEOS is VERY popular and its absence in Amazon Linux dnf requires either inconvenient workarounds or custom AMI setup, which in both cases end up costing a lot of time and effort for customers, and this could be easily solved and save a lot of resources if this widely used package was available in the AL2023 package repository

**Comment by acrolink** [comment]
The solution is just not to use Amazon Linux 2023.

**Comment by januszm** [comment]
That's not a solution. One way is to compile the version of geos that you need for AL2023. Package as zip or tar.gz and store on s3, then pull it with aws a3 command line tool that is available on each AL and unpack. It's fast  but obviously it'd be better if it came in a form of an rpm package avaliable via yum/dnf

**Comment by acrolink** [comment]
It seem will take me 2 hours to install this thing with all the prerequisites, instead of 5 seconds it could have taken in Ubuntu. I could have set up a fresh Ubuntu machine in that time with Postgis. So for me, no more Amazon Linux, neither 2023 nor 2030.

**Comment by ThrawnCA** [comment]
> The solution is just not to use Amazon Linux 2023.

This might have been a reasonable comment in other contexts, but not as a comment _on the Amazon Linux 2023 repository_. It's not helpful or constructive here. Take it elsewhere.

**Comment by januszm** [comment]
@acrolink definitely not 3-4 hours, and you only need to do it once for the geos version of your choice. SSH to one of your running  AL2023 servers, download GEOS source and compile and make install to a subdirectory.  Compress (e.g. `tar cjf`) and upload to an S3 bucket.

Then add this to `.ebextensions` if you're using Elastic Beanstalk, or to your own deployment script where you would normally put `yum/dnf install geos geos-devel`

```
commands:
  install_geos:
    command: |
      aws s3 cp s3://YOUR_S3_BUCKETs/geos-al2023/geos.tar.bz2 geos.tar.bz2
      tar xfo geos.tar.bz2 -C /usr/
```
and voila. The actual side benefit of this solution is that your server no longer depends on an external server such as fedora or other rpm source.
But yes, still I would like to see `geos` and `geos-devel` packages as part of Amazon Linux packages repository.

**Comment by acrolink** [comment]
@januszm Thank you. I actually needed `geos` to install `postgis`. But I believe same method you had suggested should also work.  Interesting to know if `postgis` can be enabled easily in AWS Aurora PostgreSQL database instances.

**Comment by januszm** [comment]
https://aws.amazon.com/about-aws/whats-new/2021/10/amazon-aurora-postgresql-supports-postgis/

**Comment by Raju-perumal** [comment]
@januszm Thank you. How about future package update ?

So we need to check periodically, compile , tar ball, upload S3 .. etc.,

**Comment by januszm** [comment]
Depends on your needs, I do it once a year

**Comment by acrolink** [comment]
I think it all comes down to the need to use Amazon Linux 2023 or not. So surely varies across needs, applications and the needed eco-system around.

**Comment by SreeSunku** [comment]
geos and geos-devel are available in AL2023 : https://docs.aws.amazon.com/linux/al2023/release-notes/all-packages-AL2023.9.html

**Comment by grantmcdermott** [comment]
@stangelandcl can we close this?
